src/cargo/util/toml_mut/dependency.rs

```diff
@@ -921,6 +928,76 @@ impl Display for WorkspaceSource {
     }
 }
 
+pub struct Feature {
+    /// The name of the feature (as it is set in its `Cargo.toml`).
+    pub name: String,
+    /// List of features to add.
+    pub features: IndexSet<String>,
+    /// this is the new name for the dependency
+    /// as a string.  None if it is not renamed.
+    pub rename: Option<String>,
+}
+
+impl Feature {
+    pub fn new(name: String, rename: Option<String>) -> Self {
+        Self {
+            name: name,
+            features: IndexSet::new(),
+            rename: rename,
+        }
+    }
+
+    pub fn add_feature(mut self, value: String) -> Self {
+        self.features.insert(value);
+        self
+    }
+
+    pub fn as_feature(&self) -> String {
+        if let Some(rename) = &self.rename {
+            format!("dep:{}", rename)
+        } else {
+            format!("dep:{}", self.name)
+        }
+    }
+
+    /// Convert feature to TOML.
+    /// Panics if the path is relative
+    pub fn to_toml(&self, crate_root: &Path) -> toml_edit::Item {
+        assert!(
+            crate_root.is_absolute(),
+            "Absolute path needed, got: {}",
+            crate_root.display()
+        );
+        let features: toml_edit::Value = self.features.iter().cloned().collect();
+        toml_edit::value(features)
+    }
+
+    /// Check whether `dep:<dep>` is defined in the features section.
+    ///
+    /// `true`  if there is a define `dep:<dep>`
+    /// `false` if `dep:<dep>` is not used ever.
+    ///
+    /// Make sure that `dep:<dep>` is included in one of features in the features table.
+    pub fn check_duplicate_feature(&self, item: &mut toml_edit::Item) -> bool {
```

Comment:
The name of this function does not match what it does.

I would expect this to either
- Check for two features of the same name (which can't happen)
- Check for duplicate activations within a single feature but we need to check for an existing activation of a feature value among all features

Reply:
thanks and Modified to `is_active_as_feature`